src/FS.Embedded.Manifest.Task/build/netstandard2.0/Microsoft.Extensions.FileSystem.Embedded.Manifest.Task.targets

```diff
@@ -0,0 +1,44 @@
+<Project>
+
+  <UsingTask
+    AssemblyFile="Microsoft.Extensions.FileSystem.Embedded.Manifest.Task"
+    TaskName="Microsoft.Extensions.FileProviders.Embedded.Manifest.Task.GenerateEmbeddedResourcesManifest" />
+
+  <PropertyGroup>
+    <EmbeddedFilesManifest Condition="$(EmbeddedFilesManifest) == ''">false</EmbeddedFilesManifest>
+    <EmbeddedFilesManifestFileName Condition="$(EmbeddedFilesManifestFileName) == ''">Microsoft.Extensions.FileSystem.Embedded.Manifest.xml</EmbeddedFilesManifestFileName>
+  </PropertyGroup>
+
+  <ItemGroup Condition="$(EmbeddedFilesManifest) == 'true'">
+    <EmbeddedResource
+      Include="$(IntermediateOutputPath)$(EmbeddedFilesManifestFileName)"
+      IsManifest="true"
+      Condition="@(EmbeddedResource->WithMetadataValue('IsManifest','true')->Count()) == 0"
+      LogicalName="$(EmbeddedFilesManifestFileName)" />
+  </ItemGroup>
+
+  <Target 
+    Name="GenerateEmbeddedFilesManifest"
+    AfterTargets="PrepareResourceNames"
+    Condition="$(EmbeddedFilesManifest) == 'true'"
+    Inputs="@(EmbeddedResource)"
```

Comment:
This is a good start. You need a few additional things to make this work correctly. Inputs/Outputs only compares the timestamps of each file and asserts that `Inputs.Max(f => f.LastModifiedTime) < Outputs.Min(f =>f .LastModifiedTime)`. What it doesn't do is compare if the list of Inputs and Outputs changes between runs. So if I remove an embedded resource, the assertion still checks out, and the manifest isn't re-built, but the list of embedded files is wrong.

The way we've worked around this in dotnet/sdk is to add an additional input, which generates a hash of all items and relevant metadata.

For a good example of this, checkout the assembly attribute generation targets: https://github.com/dotnet/sdk/blob/2995c8495082d1dd4bcad35c37007375edf45c6f/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.GenerateAssemblyInfo.targets#L114

```xml 
  <!-- 
    To allow version changes to be respected on incremental builds (e.g. through CLI parameters),
    create a hash of all assembly attributes so that the cache file will change with the calculated
    assembly attribute values and msbuild will then execute CoreGenerateAssembly to generate a new file.
  -->
  <Target Name="CreateGeneratedAssemblyInfoInputsCacheFile"
          DependsOnTargets="GetAssemblyAttributes">
    <PropertyGroup>
      <GeneratedAssemblyInfoInputsCacheFile>$(IntermediateOutputPath)$(MSBuildProjectName).AssemblyInfoInputs.cache</GeneratedAssemblyInfoInputsCacheFile>
    </PropertyGroup>

    <!-- We only use up to _Parameter1 for most attributes, but other targets may add additional assembly attributes with multiple parameters. -->
    <Hash ItemsToHash="@(AssemblyAttribute->'%(Identity)%(_Parameter1)%(_Parameter2)%(_Parameter3)%(_Parameter4)%(_Parameter5)%(_Parameter6)%(_Parameter7)%(_Parameter8)')">
      <Output TaskParameter="HashResult" PropertyName="_AssemblyAttributesHash" />
    </Hash>

    <WriteLinesToFile Lines="$(_AssemblyAttributesHash)" File="$(GeneratedAssemblyInfoInputsCacheFile)" Overwrite="True" WriteOnlyWhenDifferent="True" />
    
    <ItemGroup>
      <FileWrites Include="$(GeneratedAssemblyInfoInputsCacheFile)" />
    </ItemGroup>
  </Target>

  <Target Name="CoreGenerateAssemblyInfo"
          Condition="'$(Language)'=='VB' or '$(Language)'=='C#'"
          DependsOnTargets="CreateGeneratedAssemblyInfoInputsCacheFile"
          Inputs="$(GeneratedAssemblyInfoInputsCacheFile)"
          Outputs="$(GeneratedAssemblyInfoFile)">
    <ItemGroup>
      <!-- Ensure the generated assemblyinfo file is not already part of the Compile sources, as a workaround for https://github.com/dotnet/sdk/issues/114 -->
      <Compile Remove="$(GeneratedAssemblyInfoFile)" />
    </ItemGroup>

    <WriteCodeFragment AssemblyAttributes="@(AssemblyAttribute)" Language="$(Language)" OutputFile="$(GeneratedAssemblyInfoFile)">
      <Output TaskParameter="OutputFile" ItemName="Compile" />
      <Output TaskParameter="OutputFile" ItemName="FileWrites" />
    </WriteCodeFragment>
  </Target>
```